src/ServiceClient.js

```diff
@@ -0,0 +1,184 @@
+import grpc, { InterceptingCall } from "grpc";
+import url from "url";
+import { BigNumber } from 'bignumber.js';
+import { find, map } from 'lodash';
+import paymentChannelStateServices from './payment_channel_state_service_grpc_pb';
+
+export default class ServiceClient {
+  constructor(sdk, metadata, group, ServiceStub, paymentChannelManagementStrategy, options = {}) {
+    this._sdk = sdk;
+    this._options = options;
+    this._metadata = metadata;
+    this._group = group;
+    this._paymentChannelManagementStrategy = paymentChannelManagementStrategy;
+    this._grpcStub = this._generateGrpcStub(ServiceStub);
+    this._paymentChannelStateServiceClient = this._generatePaymentChannelStateServiceClient();
+    this._paymentChannels = [];
+  }
+
+  get stub() {
+    return this._grpcStub;
+  }
+
+  get group() {
+    return this._group;
+  }
+
+  get groupIdInBytes() {
+    return Buffer.from(this.group.group_id, 'base64');
+  }
+
+  get paymentChannels() {
+    return this._paymentChannels;
+  }
+
+  get metadata() {
+    return this._metadata;
+  }
+
+  get paymentAddress() {
```

Comment:
I suggest removing this method as it can be replaced by chain of calls `group().payment_address`

Reply:
I think having them as convenience methods just makes it easier for the developer to have easy access to the information rather than figuring our what all is there in the `metadata` and `group`. My thought process was to make it as simple as possible for a developer to get started with it. This is for all the convenience methods that have been added across the code base.

Reply:
The reason I would move such methods to `group()` and `metadata()` is the "principle of least surprise". All information about service like: price, payment address, list of groups should be available from single place and it is natural to have them available from `metadata()`. Similarly group name and list of endpoints and groupId should be avialable from `group()`. Thus user will need some background and knowledge about what is `metadata` or `group` but as soon as he have this background it will easily navigate through this API. 

If we make some specific information like `paymentAddress` being available from another object, then it will be surprising for user of API why `price` is not available from the same place.

Reply:
Agree. Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/1/commits/78cf9ae1c7ce7ce7b1812ea3d016dc3691663314)